### PR TITLE
try pinning numpy dev

### PR DESCRIPTION
**Pull request by TomAugspurger**
xref #22960

### CODE REVIEW

File: ci/travis-37-numpydev.yaml [hunk 1]
```diff
@@ -13,5 +13,5 @@ dependencies:
     - "git+git://github.com/dateutil/dateutil.git"
     - "-f https://7933911d6844c6c53a7d-47bd50c35cd79bd838daf386af554a83.ssl.cf2.rackcdn.com"
     - "--pre"
-    - "numpy"
+    - "numpy<=1.16.0.dev0+20181015190246"
```
Comment (jreback) [inline ci/travis-37-numpydev.yaml]: these are just strings, so the < part won't work here, just pin = for now

Comment (TomAugspurger) [reply ci/travis-37-numpydev.yaml]: That grabbed NmPy commit 54985e3, so it seemed to have worked.

Comment (TomAugspurger) [reply ci/travis-37-numpydev.yaml]: https://travis-ci.org/pandas-dev/pandas/jobs/442114100#L727

**Comment by h-vetinari** [comment]
Something with `pyarrow` also started failing overnight (the azure Win36 job is failing for all recent builds: https://dev.azure.com/pandas-dev/pandas/_build - also happened for this PR, but for some reason the azure link is not visible in the results anymore; here it is): https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=1251&view=logs

**Pull request merged**